neo/public/javascripts/map/controllers.js

```diff
@@ -93,9 +96,17 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common'])
       leafletData.getMap().then(function(map) {
         $scope.map = map;
         $scope.bounds = map.getBounds();
+
+        $scope.tree = rbush();
+        $scope.cache ;
+        $scope.cachecount = 0;
+        $scope.rm_duplicate = new Set();
+       
```

Comment:
I think the `cache` deserves a separate data structure and the `tree` and `cachecount` are its attribute.